src/node_crypto.cc

```diff
@@ -4674,38 +4550,21 @@ class PBKDF2Request : public AsyncWrap {
   PBKDF2Request(Environment* env,
                 Local<Object> object,
                 const EVP_MD* digest,
-                int passlen,
-                char* pass,
-                int saltlen,
-                char* salt,
-                int iter,
-                int keylen)
+                MallocedBuffer<char>&& pass,
+                MallocedBuffer<char>&& salt,
```

Comment:
Coincidence: I was working yesterday on simplifying this to something that uses `std::vector<char>`. 

(Working on adding scrypt support and DRYing the PBKDF2 and RandomBytes code in the process.)